tests/test_trx.py

```diff
@@ -590,3 +590,66 @@ def test_trx_unknown_trc20_send(self, backend, configuration, firmware,
                     "a9059cbb000000000000000000000000364b03e0815687edaf90b81ff58e496dea7383d700000000000000000000000000000000000000000000000000000000000f4240"
                 )))
         self.sign_and_validate(client, firmware, 0, tx)
+
+    def test_trx_freezeV2_balance(self, backend, configuration, firmware,
+                                   navigator):
+        client = TronClient(backend, firmware, navigator)
+        tx = client.packContract(
+            tron.Transaction.Contract.FreezeBalanceV2Contract,
+            contract.FreezeBalanceV2Contract(
+                owner_address=bytes.fromhex(
+                    client.getAccount(0)['addressHex']),
+                frozen_balance=100000000,
+                resource=contract.ENERGY))
+        self.sign_and_validate(client, firmware, 0, tx)
+
+    def test_trx_unfreezeV2_balance(self, backend, configuration, firmware,
+                                   navigator):
+        client = TronClient(backend, firmware, navigator)
+        tx = client.packContract(
+            tron.Transaction.Contract.UnfreezeBalanceV2Contract,
+            contract.UnfreezeBalanceV2Contract(
+                owner_address=bytes.fromhex(
+                    client.getAccount(0)['addressHex']),
+                unfreeze_balance=100000000,
+                resource=contract.ENERGY))
+        self.sign_and_validate(client, firmware, 0, tx)
+
+    def test_trx_delegate_resource(self, backend, configuration, firmware,
+                                   navigator):
+        client = TronClient(backend, firmware, navigator)
+        tx = client.packContract(
+            tron.Transaction.Contract.DelegateResourceContract,
+            contract.DelegateResourceContract(
+                owner_address=bytes.fromhex(
+                    client.getAccount(0)['addressHex']),
+                resource=contract.ENERGY,
+                balance=100000000,
+                receiver_address=bytes.fromhex(
+                    client.address_hex("TGQVLckg1gDZS5wUwPTrPgRG4U8MKC4jcP")),
+                lock=0))
+        self.sign_and_validate(client, firmware, 0, tx)
+
+    def test_trx_undelegate_resource(self, backend, configuration, firmware,
```

Comment:
@ClarkChenc you will have to execute the tests locally with : 

```
pip install --extra-index-url https://test.pypi.org/simple/ -r requirements.txt
cd tests
pytest -v --device all --golden_run
```
And then push the new snapshots generated in `tests/snapshots`

Reply:
Snaphots have been added. I find ledger-secure-sdk is not easy to use, especially stax. master branch doesn't always work.